presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java

```diff
@@ -646,11 +651,29 @@ private ConnectorTableMetadata doGetTableMetadata(ConnectorSession session, Sche
     private List<String> listSchemas(ConnectorSession session, Optional<String> schemaName)
     {
         if (schemaName.isPresent()) {
+            if (!filterSchema(schemaName.get())) {
+                return ImmutableList.of();
+            }
             return ImmutableList.of(schemaName.get());
         }
         return listSchemaNames(session);
     }
 
+    private static boolean filterSchema(String schemaName)
```

Comment:
I would put this method under the last usage.

Reply:
right